src/main/java/com/hazelcast/aws/EcsClientStrategy.java

```diff
@@ -0,0 +1,113 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.aws;
+
+import com.hazelcast.aws.impl.Constants;
+import com.hazelcast.aws.impl.DescribeNetworkInterfaces;
+import com.hazelcast.aws.impl.DescribeTasks;
+import com.hazelcast.aws.impl.ListTasks;
+import com.hazelcast.aws.utility.Environment;
+import com.hazelcast.aws.utility.MetadataUtils;
+import com.hazelcast.internal.json.Json;
+import com.hazelcast.internal.json.JsonObject;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.hazelcast.aws.impl.Constants.AWS_EXECUTION_ENV_VAR_NAME;
+import static com.hazelcast.aws.impl.Constants.EC2_PREFIX;
+import static com.hazelcast.aws.impl.Constants.HTTPS;
+import static com.hazelcast.aws.utility.StringUtils.isNotEmpty;
+
+/**
+ *
+ */
+class EcsClientStrategy extends AwsClientStrategy {
+
+    private static final String UPPER_ECS = "ECS";
+
+    private static final ILogger LOGGER = Logger.getLogger(AwsClientStrategy.class);
+
+    private String metadataClusterName;
+    private String metadataFamilyName;
+    private String endpointDomain;
+
+    EcsClientStrategy(AwsConfig awsConfig, String endpoint) {
+        super(awsConfig, endpoint);
+        this.endpointDomain = endpoint.substring(Constants.HOSTNAME_PREFIX_LENGTH);
+    }
+
+    @Override
+    public Collection<String> getPrivateIpAddresses() throws Exception {
+        retrieveAndParseMetadata();
+        ListTasks listTasks = new ListTasks(awsConfig, new URL(HTTPS, endpoint, -1, "/"));
+        Collection<String> taskArns = listTasks.execute(metadataClusterName, metadataFamilyName);
+        if (!taskArns.isEmpty()) {
+            DescribeTasks describeTasks = new DescribeTasks(awsConfig, new URL(HTTPS, endpoint, -1, "/"));
+            Collection<String> taskAddresses = describeTasks.execute(taskArns, metadataClusterName, metadataFamilyName);
+            return taskAddresses;
+        }
+        return Collections.EMPTY_LIST;
+    }
+
+    @Override
+    @SuppressWarnings(value = "unchecked")
+    public Map<String, String> getAddresses() throws Exception {
+        retrieveAndParseMetadata();
+        ListTasks listTasks = new ListTasks(awsConfig, new URL(HTTPS, endpoint, -1, "/"));
+        Collection<String> taskArns = listTasks.execute(metadataClusterName, metadataFamilyName);
+        if (!taskArns.isEmpty()) {
+            DescribeTasks describeTasks = new DescribeTasks(awsConfig, new URL(HTTPS, endpoint, -1, "/"));
+            Collection<String> taskAddresses = describeTasks.execute(taskArns, metadataClusterName, metadataFamilyName);
+            DescribeNetworkInterfaces describeNetworkInterfaces =
+                    new DescribeNetworkInterfaces(awsConfig, new URL(HTTPS, EC2_PREFIX + endpointDomain, -1, "/"));
```

Comment:
Consider changing the constructor parameters of all these `Describe...` classes to be a little high level here. Like I'd imagine the execution here as:
```
DescribeNetworkInterfaces describeNetworkInterfaces = new DescribeNetworkInterfaces(awsConfig, endpointDomain);
```